src/Styleguide/Elements/Responsive.tsx

```diff
@@ -0,0 +1,85 @@
+import React from "react"
+
+const ResponsiveContext = React.createContext({})
+
+interface Breakpoints {
+  [breakpoint: string]: string
+}
+
+interface ResponsiveProviderProps {
+  breakpoints: Breakpoints
+}
+
+interface BreakpointState {
+  [breakpoint: string]: boolean
+}
+
+interface ResponsiveProviderState {
+  breakpoints: BreakpointState
+  mediaMatchers: MediaQueryList[]
+  mqHandler: null | MediaQueryListListener
+}
+
+export class ResponsiveProvider extends React.Component<
+  ResponsiveProviderProps,
+  ResponsiveProviderState
+> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      breakpoints: {
+        ...Object.keys(props.breakpoints)
+          .map(breakpoint => ({
+            [breakpoint]: false,
+          }))
+          .reduce((acc, curr) => ({ ...acc, ...curr }), {}),
+      },
+      mediaMatchers: [],
+      mqHandler: null,
+    }
+  }
+
+  componentDidMount() {
+    this.setupObservers(this.props.breakpoints)
+  }
+
+  componentWillUnmount() {
```

Comment:
Heads up that this lifecycle is [going away](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html)

Reply:
Actually, scratch that...